src/core/Modal/Modal/Modal.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { Component, ReactNode, MouseEvent } from 'react';
+import ReactDOM from 'react-dom';
+import { default as styled } from 'styled-components';
+import { createFocusTrap, FocusTrap } from 'focus-trap';
+import classnames from 'classnames';
+import { HtmlDiv, HtmlDivProps } from '../../../reset';
+import { AutoId } from '../../../utils/AutoId';
+import { windowAvailable } from '../../../utils/common';
+import { ModalContent, ModalFooter } from '../';
+import './Modal.globals.scss';
+import { baseStyles } from './Modal.baseStyles';
+
+export type ModalVariant = 'smallScreen' | 'default';
+
+export interface ModalProps
+  extends Omit<HtmlDivProps, 'children' | 'className' | 'title' | 'ref'> {
+  /** Show or hide the modal */
+  visible: boolean;
+  /** Modal container div class name for custom styling. */
+  className?: string;
+  /** Children */
+  children: ModalContent | ModalFooter | ReactNode;
+  /**
+   * Variant. Use smallScreen for mobile and small displays
+   * @default 'default'
+   */
+  variant?: ModalVariant;
+  /**
+   * Show vertical scroll bar if needed and show horizontal divider before buttons.
+   * @default true
+   */
+  scrollable?: boolean;
+  /**
+   * Use portal instead of injecting content inside the Modal component
+   * @default true
+   * Portal is never used when rendering on server or modal mount node is not available.
```

Comment:
Needs another 'when'. Corrected version:

Portal is never used when rendering on server or when modal mount node is not available.

Reply:
Fixed.